File: saucedemo-tests/pages/InventoryPage.js
class InventoryPage {
  constructor(page) {
    this.page = page;
    this.backpackTitle = page.locator('#item_4_title_link');
    this.addToCartBackpackBtn = page.locator('[data-test="add-to-cart-sauce-labs-backpack"]');
    this.cartBadge = page.locator('[data-test="shopping-cart-badge"]');
    this.cartLink = page.locator('[data-test="shopping-cart-link"]');
  }

  // Path A: Пряме додавання з головної сторінки
  async addBackpackToCart() {
    await this.addToCartBackpackBtn.click();
  }

  // Path B: Клік на назву для переходу в картку товару
  async openBackpackDetails() {
    await this.backpackTitle.click();
  }

  async clickCart() {
    await this.cartLink.click();
  }
}

module.exports = { InventoryPage };

File: saucedemo-tests-olena/pages/inventory_page.py
class InventoryPage:
    def __init__(self, page):
        self.page = page
        self.backpack_title = page.locator('[data-test="item-4-title-link"]')
        self.backpack_price = page.locator('[data-test="inventory-item-price"]').first
        self.backpack_image = page.locator('[data-test="inventory-item-sauce-labs-backpack-img"]')
        self.cart_link = page.locator('[data-test="shopping-cart-link"]')

    def open_backpack_details(self):
        self.backpack_title.click()

    def get_backpack_name(self):
        return self.backpack_title.inner_text()

    def get_backpack_price(self):
        return self.backpack_price.inner_text()

    def get_backpack_image_src(self):
        return self.backpack_image.get_attribute("src")

    def open_cart(self):
        self.cart_link.click()
